src/EnergyPlus/DataZoneEquipment.cc

```diff
@@ -312,106 +312,488 @@ void GetZoneEquipmentData(EnergyPlusData &state)
+        state.dataZoneEquip->ReturnAirPath(PathNum).ComponentTypeEnum.allocate(state.dataZoneEquip->ReturnAirPath(PathNum).NumOfComponents);
+        state.dataZoneEquip->ReturnAirPath(PathNum).ComponentTypeEnum = DataZoneEquipment::AirLoopHVACZone::Invalid;
+        state.dataZoneEquip->ReturnAirPath(PathNum).ComponentName.allocate(state.dataZoneEquip->ReturnAirPath(PathNum).NumOfComponents);
+        state.dataZoneEquip->ReturnAirPath(PathNum).ComponentIndex.allocate(state.dataZoneEquip->ReturnAirPath(PathNum).NumOfComponents);
+
+        int Counter = 3;
+
+        for (int CompNum = 1; CompNum <= state.dataZoneEquip->ReturnAirPath(PathNum).NumOfComponents; ++CompNum) {
+
+            if ((AlphArray(Counter) == "AIRLOOPHVAC:ZONEMIXER") || (AlphArray(Counter) == "AIRLOOPHVAC:RETURNPLENUM")) {
+
+                state.dataZoneEquip->ReturnAirPath(PathNum).ComponentType(CompNum) = AlphArray(Counter);
+                state.dataZoneEquip->ReturnAirPath(PathNum).ComponentName(CompNum) = AlphArray(Counter + 1);
+                state.dataZoneEquip->ReturnAirPath(PathNum).ComponentIndex(CompNum) = 0;
+                ValidateComponent(state,
+                                  state.dataZoneEquip->ReturnAirPath(PathNum).ComponentType(CompNum),
+                                  state.dataZoneEquip->ReturnAirPath(PathNum).ComponentName(CompNum),
+                                  IsNotOK,
+                                  CurrentModuleObject);
+                if (IsNotOK) {
+                    ShowContinueError(state, format("In {} = {}", CurrentModuleObject, state.dataZoneEquip->ReturnAirPath(PathNum).Name));
+                    state.dataZoneEquip->GetZoneEquipmentDataErrorsFound = true;
+                }
+                state.dataZoneEquip->ReturnAirPath(PathNum).ComponentTypeEnum(CompNum) =
+                    static_cast<AirLoopHVACZone>(getEnumValue(AirLoopHVACTypeNamesUC, AlphArray(Counter)));
+            } else {
+                ShowSevereError(state, format("{}{}=\"{}\"", RoutineName, cAlphaFields(1), state.dataZoneEquip->ReturnAirPath(PathNum).Name));
+                ShowContinueError(state, format("Unhandled component type =\"{}\".", AlphArray(Counter)));
+                ShowContinueError(state, R"(Must be "AirLoopHVAC:ZoneMixer" or "AirLoopHVAC:ReturnPlenum")");
+                state.dataZoneEquip->GetZoneEquipmentDataErrorsFound = true;
+            }
+
+            Counter += 2;
+        }
+
+    } // end loop over return air paths
+
+    AlphArray.deallocate();
+    NumArray.deallocate();
+    cAlphaFields.deallocate();
+    cNumericFields.deallocate();
+    lAlphaBlanks.deallocate();
+    lNumericBlanks.deallocate();
+
+    if (state.dataZoneEquip->GetZoneEquipmentDataErrorsFound) {
+        ShowFatalError(state, format("{}Errors found in getting Zone Equipment input.", RoutineName));
+    }
+}
+
+void processZoneEquipmentInput(EnergyPlusData &state,
```

Comment:
Move ZoneHVAC:EquipmentConnections input processing to a new function so it can be re-used for SpaceHVAC:EquipmentConnections.